proposal/package-importer.d.ts.md

```diff
@@ -0,0 +1,386 @@
+
+[conditional exports]: https://nodejs.org/api/packages.html#conditional-exports
+[Vite]: https://github.com/vitejs/vite/pull/7817
+[Parcel]:
+    https://github.com/parcel-bundler/parcel/blob/2d2400ded4615375ee6bd53ef77b4857ad1591dd/packages/transformers/sass/src/SassTransformer.js#L163
+[Sass Loader for Webpack]:
+    https://github.com/webpack-contrib/sass-loader/blob/02df41203adfda96959e56abb43bd35a89ec11ba/src/utils.js#L514
+
+Because use of conditional exports is flexible and recommended for modern
+packages, this will be the primary method used for the Node package importer. We
+will support both the `"sass"` and the `"style"` conditions, as Sass can also
+use the CSS exports exposed through `"style"`. While in practice, `"style"`
+tends to be used solely for `css` files, we will support `scss`, `sass` and
+`css` files for either `"sass"` or `"style"`.
+
+## Types
+
+```ts
+import '../spec/js-api';
+```
+
+#### `usePkgImporter`
+
+If true, the compiler will use the built-in package importer to resolve any url
+with the `pkg` scheme. This importer follows Node logic to locate Sass files.
+
+Defaults to false.
+
+```ts
+declare module '../spec/js-api/options' {
+  interface Options<sync extends 'sync' | 'async'> {
+    /**
+     * Whether or not to enable the built-in package importer to resolve any url
+     * with the `pkg` scheme. This importer follows Node.js resolution logic.
+     *
+     * @defaultValue `false`
+     * @category Input
+     */
+    usePkgImporter?: boolean;
+  }
+}
+```
+
+## Semantics
+
+### Package Importers
+
+This proposal defines the requirements for Package Importers written by users or
+provided by implementations. It is a type of [Filesystem Importer] and will
+handle non-canonical URLs:
+
+- with the scheme `pkg`
+- followed by a package name
+- optionally followed by a path, with path segments separated with a forward
+  slash.
+
+The package name will often be the first path segment, but the importer should
+take into account any conventions in the environment. For instance, Node
+supports scoped package names, which start with `@` followed by 2 path segments.
+
+[filesystem importer]: ../spec/modules.md#filesystem-importer
+
+### Node Specific Semantics
+
+The Node package importer is an [importer] with an associated absolute `pkg:`
+URL named `base`. When the Node package importer is invoked with a string named
+`string` and a [previous url] `previousUrl`:
+
+- Let `url` be the result of [parsing `string` as a URL][parsing a url] with
+  `base` as the base URL. If this returns a failure, throw that failure.
+- If `url`'s scheme is not `pkg`, return null.
+- Let `resolved` be the result of [resolving a `pkg` URL] with `url` and `previousURL`
+- If `resolved` is null, return null.
+- Let `text` be the contents of the file at `resolved`.
+- Let `syntax` be:
+  - "scss" if `url` ends in `.scss`.
+  - "indented" if `url` ends in `.sass`.
+  - "css" if `url` ends in `.css`.
```

Comment:
Shouldn't these be checking `resolved`? `url` might be `pkg:bootstrap` which doesn't have any extension at all.